src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,7 +579,7 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
 
         private void OnEnvDTEProjectAdded(EnvDTE.Project envDTEProject)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
```

Comment:
I ran E2E tests, and they are passing. Even LSL tests are also passing with my second fix for PMC (which I'm going to raise soon). So I'll suggest we continue with this PR and let it run it run these E2E several times as part of CI and then we'll evaluate if it still repro's some hangs there. Sounds good?